edge-hub/src/Microsoft.Azure.Devices.Routing.Core/RouteFactory.cs

```diff
@@ -25,11 +31,14 @@ protected RouteFactory(IEndpointFactory endpointFactory)
 
         public abstract string GetNextRouteId();
 
-        public Route Create(string routeString)
+        public Route Create(string routeString) =>
+            this.Create(routeString, DefaultPriority, 0);
```

Comment:
why default TTL is zero?  Does this mention in the spec?

Reply:
0 is treated as a special value for TTL, there's no valid scenario where any message should have 0sec TTL for real.  We use zero to signify that the operator did not specify a TTL on the route, at which point the global TTL value should take over.

I'll add a comment to explain this.

---
In reply to: [377458687](https://github.com/Azure/iotedge/pull/2469#discussion_r377458687) [](ancestors = 377458687)